node/src/types/block/hot_block/state.rs

```diff
@@ -40,6 +40,7 @@ pub(crate) struct State {
     pub(super) tried_to_sign: bool,
     pub(super) sent_to_consensus_post_execution: bool,
     pub(super) sent_to_accumulator_post_execution: bool,
+    pub(super) sent_to_synchronizer_post_execution: bool,
```

Comment:
I don't think we need this new field.  Seems like we always want to send `block_synchronizer::Event::ExecutedBlock` to the synchronizer after the block is marked complete by the accumulator (similar to the event stream server events which are created after the block is marked complete).

Reply:
I don't think this is true because once we sent the event once, the state transitions in the synchronizer would put it into the `Synced` state, and subsequently the builder will be purged on the next `keep_up_instruction` call.

In case we allow duplicates and for some reason they actually happen, they will either change the `executing_progress` state change to `Done` a second time (with no additional effect) or miss the call because there is no builder there. Either way seems like a superfluous event to me.

I understand this looks like it's polluting the hot block state structure, but with its current design it looks like that's what it's there for; to keep relevant state changes across components. Is there another reason you'd want this removed that I didn't mention in this comment?

Reply:
As discussed offline, we want to have this check to enforce no duplicate events are dispatched, but it makes sense to apply that to all the subsequent events that happen in `handle_hot_block`. As agreed, we now have an `all_actions_done` flag in the state instead of `sent_to_synchronizer_post_execution` which guards the rest of the events as well.